jina/serve/networking.py

```diff
@@ -393,73 +442,40 @@ async def send_requests(
-                        requests[0].nbytes
-                    )
-                    with MetricsTimer(
-                        self._metrics.sending_requests_time_metrics,
-                        self._histograms.sending_requests_time_metrics,
-                        self._histograms.histogram_metric_labels,
-                    ):
+                    self._record_request_bytes_metric(requests[0].nbytes)
+                    with timer:
                         metadata, response = (
                             await call_result.trailing_metadata(),
                             await call_result,
                         )
-
-                        if self._metrics.received_response_bytes:
-                            self._metrics.received_response_bytes.observe(
-                                response.nbytes
-                            )
-                        self._histograms.record_received_response_bytes(response.nbytes)
+                        self._record_received_bytes_metric(response.nbytes)
                     return response, metadata
 
                 elif self.stream_stub:
                     for response in requests:
-                        if self._metrics.send_requests_bytes_metrics:
-                            self._metrics.send_requests_bytes_metrics.observe(
-                                response.nbytes
-                            )
-                        self._histograms.record_send_requests_bytes_metrics(
-                            response.nbytes
-                        )
+                        self._record_request_bytes_metric(response.nbytes)
 
-                    with MetricsTimer(
-                        self._metrics.sending_requests_time_metrics,
-                        self._histograms.sending_requests_time_metrics,
-                        self._histograms.histogram_metric_labels,
-                    ):
+                    with timer:
                         async for response in self.stream_stub.Call(
                             iter(requests),
                             compression=compression,
                             timeout=timeout,
                             metadata=metadata,
                         ):
-                            if self._metrics.received_response_bytes:
-                                self._metrics.received_response_bytes.observe(
-                                    response.nbytes
-                                )
-                            self._histograms.record_received_response_bytes(
-                                response.nbytes
-                            )
-
+                            self._record_received_bytes_metric(response.nbytes)
                             return response, None
+
             if request_type == DataRequest and len(requests) > 1:
+                # TODO: This section of the code has no observability
+                # TODO: Does this section not need to deal with streaming stubs?
```

Comment:
I think this should be handled as well, perhaps open another ticket and see if it needs handling.

I believe this code is reached when there is a Flow like this:

```python
f = Flow().add(name='name1').add(name='name2', needs=['gateway']).add(name='name3', needs=['name1', 'name2'], disable_reduce=True)
```

then Gateway needs to send 2 requests to `name3` without reducing before.

Reply:
Ticket opened: https://github.com/jina-ai/jina/issues/5359